### ISSUE TITLE
docker-machine generic driver workflow question

### ISSUE THREAD

**Issue by gngeorgiev** [docker/machine#2558]
Hello,

I have a question regarding docker-machine and its generic driver in the following case: Lets say I configure 4 machines from a single machine, docker-machine stores the data and certificates for these machines. What should happen if the machine running docker-machine needs to be recreated and all the settings and certs are lost. I do not think that running `create` again would be good, since it will provision the host again, restarting the docker daemon etc.

Excuse me in advance if I am missing something.

Suggestions are highly appreciated.

**Comment by jeanlaurent** [comment]
As of now, the machine will not be completely re-provision, but at least the certs will be changed and the docker-daemon re-created.

I believe if you do this by hand (ie, provision without docker-machine), and loose the certs, you'll be in a similar situation.

**Comment by gngeorgiev** [comment]
Yes, I agree that without docker-machine we would end up in a similar situation.

In this case, can we backup the `~/.docker/machine/{certs,machines}` folders?

**Comment by omallo** [comment]
@jeanlaurent : I have a somewhat related question: I have a scenario where I created some docker hosts using docker-machine from my laptop. When I created the hosts, I used my own TLS certificates (CA certificate/key and client certificate/key). Now, I would like to start managing the same hosts from another PC where I have the same TLS certificates available. Is it possible to use docker-machine's generic driver to attach to my existing hosts without my docker engines being restarted? In such a scenario I see no real reason why docker-machine should restart the docker engines since nothing really changed (the TLS certificates need not be recreated).
